### PR TITLE
Fix compilation for zig master

### PR DESCRIPTION
**Pull request by iddev5**
This PR fixes bog's compilation for Zig master 0.9.0-dev.897+81e2034d4 (os reorganization).

Additional changes:
- AutoIndentingStream taken from std/zig/render.zig
- Zig's std.math.sqrt() no longer works on i64. Bog's std lib casts it the argument into a u64.

All tests and examples are working on my PC (having Linux Mint/Ubuntu 20.04 x86_64)

### CODE REVIEW

File: src/tokenizer.zig [hunk 1]
```diff
@@ -228,7 +231,7 @@ pub const Token = struct {
         .{ "_", .Underscore },
     });
 
-    pub fn string(id: @TagType(Id)) []const u8 {
+    pub fn string(id: std.meta.TagType(Id)) []const u8 {
```
Comment (Vexu) [inline src/tokenizer.zig:234]: Same here.

File: src/value.zig [hunk 2]
```diff
@@ -244,7 +261,7 @@ pub const Value = union(Type) {
                 .num => |b_val| n == b_val,
                 else => false,
             },
-            else => if (a.* != @as(@TagType(@TypeOf(b.*)), b.*)) return false,
+            else => if (a.* != @as(std.meta.TagType(@TypeOf(b.*)), b.*)) return false,
```
Comment (Vexu) [inline src/value.zig:264]: And here.

File: src/value.zig [hunk 3]
```diff
@@ -259,7 +276,7 @@ pub const Value = union(Type) {
                 }
                 return true;
             },
-            .map => |*m| @panic("TODO eql for maps"),
+            .map => |_| @panic("TODO eql for maps"),
```
Comment (Vexu) [inline src/value.zig:279]:
```suggestion
            .map => @panic("TODO eql for maps"),
```

File: src/vm.zig [hunk 4]
```diff
@@ -172,54 +172,39 @@ pub const Vm = struct {
                     const rhs = try vm.getNum(inst.triple.rhs);
 
                     // TODO https://github.com/ziglang/zig/issues/3234 on all of these
-                    const copy: Value = if (needNum(lhs, rhs))
-                        .{ .num = asNum(lhs) + asNum(rhs) }
-                    else
-                        .{ .int = lhs.int + rhs.int };
+                    const copy: Value = if (needNum(lhs, rhs)) .{ .num = asNum(lhs) + asNum(rhs) } else .{ .int = lhs.int + rhs.int };
```
Comment (Vexu) [inline src/vm.zig:175]: Add commas to all of these to avoid `zig fmt` creating long lines.

**Review by Vexu — CHANGES_REQUESTED** [review]
Few tweaks needed but otherwise looks good.

**Comment by iddev5** [comment]
Made the changes, hope everything is cleared! Let me know if something is off

**Comment by Vexu** [comment]
Thanks!

**Pull request merged**